### DISCUSSION TITLE
20240313 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#86] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - 
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x
- MVanDijke(@mickvandijke) - 
- Iván Santana(@ivan9989-pkj) - x
- Ismael Torrado(@Lepisma07) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Developer

#### Torrust Tracker

Modify the `torrents` API endpoint to get statistics for a list of torrents:

http://localhost:1212/api/v1/torrents?token=MyAccessToken&info_hash=f584cba7dd4008ecc026ac2dc0ce1ad179822f5f&info_hash=f59caeaf12e7bc8a289c39b698d085bc27eec1c2

Like the `scrape` tracker request.

Sample response:

```json
[
  {
    "info_hash": "f584cba7dd4008ecc026ac2dc0ce1ad179822f5f",
    "seeders": 1,
    "completed": 0,
    "leechers": 0
  },
  {
    "info_hash": "f59caeaf12e7bc8a289c39b698d085bc27eec1c2",
    "seeders": 1,
    "completed": 0,
    "leechers": 0
  }
]
```

#### Torrust Index

**Bugs:**

Fix a bug I introduced in a new DB migration recently. The bug truncated the torrent secondary tables.

- https://github.com/torrust/torrust-index/issues/526

**Issues:**

Improve the importation of torrent stats from the tracker. Importation in batches and using the new tracker API endpoint feature to import a list of torrents in a single request.

- https://github.com/torrust/torrust-index/issues/469

#### Torrust Index GUI

Issues:

- https://github.com/torrust/torrust-index-gui/issues/496 EPIC (Canonical InfoHash)
  - https://github.com/torrust/torrust-index-gui/issues/242
  - https://github.com/torrust/torrust-index-gui/issues/279

### Operative (5,5h)

#### Engineer Manager

- One one-on-one

#### Tech Lead

- First meeting with new internship students.

**Comment by mario-nt** [comment]
### GUI

- https://github.com/torrust/torrust-index-gui/issues/401
- https://github.com/torrust/torrust-index-gui/pull/501

### Index
	
- https://github.com/torrust/torrust-index/issues/448
- https://github.com/torrust/torrust-index/pull/537	

### Discussions

- https://github.com/torrust/torrust-index/pull/525
- https://github.com/torrust/torrust-index/issues/424#event-11757385477
- https://github.com/torrust/torrust-index/pull/527#event-12057494316
- https://github.com/torrust/torrust-index/discussions/356#discussioncomment-8721760
- https://github.com/torrust/torrust-index/issues/526
- https://github.com/torrust/torrust-index/discussions/528

### Training/Research

- Axum 101, extensions, state and substates,
- Tower 101
- Rust traits
- Managin rust dependencies with cargo

**Comment by grmbyrn** [comment]
## Website
* [fix CodeBlock, add AnimatedHamburger.story.svelte, fix bugs #150](https://github.com/torrust/torrust-website/pull/150)
* Updating dependencies
* Learning about Histoire

## Torrust GUI
* Reading about [Torrust project](https://github.com/torrust/torrust-index-gui)

## Learning
* Vue project - [Vue expenses tracker](https://github.com/grmbyrn/vue-expenses-tracker)
